### ISSUE TITLE
CNTK BrainScript error in openning dataset file

### ISSUE THREAD

**Issue by sjsy** [Microsoft/CNTK#2986]
Hi
I want to run the examples on my system, but it cannot read the mnist dataset.
Dataset files are downloaded it the proper path.

`seyyed@seyyed-Lenovo-Y720-15IKB:~/Downloads/cntk/cntk/bin$` ./cntk configFile=~/Downloads/cntk/Examples/Image/GettingStarted/01_OneHidden.cntk RootDir=/home/seyyed/Downloads/cntk 
CNTK 2.4 (HEAD a70455, Jan 31 2018 15:10:33) at 2018/02/24 01:56:01

./cntk  configFile=/home/seyyed/Downloads/cntk/Examples/Image/GettingStarted/01_OneHidden.cntk  RootDir=/home/seyyed/Downloads/cntk
-------------------------------------------------------------------
Build info: 

		Built time: Jan 31 2018 15:03:41
		Last modified date: Tue Jan 30 03:26:13 2018
		Build type: release
		Build target: GPU
		With 1bit-SGD: no
		With ASGD: yes
		Math lib: mkl
		CUDA version: 9.0.0
		CUDNN version: 7.0.4
		Build Branch: HEAD
		Build SHA1: a70455c7abe76596853f8e6a77a4d6de1e3ba76e
		MPI distribution: Open MPI
		MPI version: 1.10.7
-------------------------------------------------------------------
-------------------------------------------------------------------
GPU info:

		Device[0]: cores = 1920; computeCapability = 6.1; type = "GeForce GTX 1060"; total memory = 6072 MB; free memory = 5684 MB
-------------------------------------------------------------------

Configuration After Processing and Variable Resolution:

configparameters: 01_OneHidden.cntk:command=trainNetwork:testNetwork
configparameters: 01_OneHidden.cntk:dataDir=/home/seyyed/Downloads/cntk/DataSets/MNIST
configparameters: 01_OneHidden.cntk:deviceId=auto
configparameters: 01_OneHidden.cntk:modelPath=./Output/Models/01_OneHidden
configparameters: 01_OneHidden.cntk:outputDir=./Output
configparameters: 01_OneHidden.cntk:precision=float
configparameters: 01_OneHidden.cntk:rootDir=/home/seyyed/Downloads/cntk
configparameters: 01_OneHidden.cntk:testNetwork={
    action = "test"
minibatchSize = 1024    
    reader = {
        readerType = "CNTKTextFormatReader"
        file = "/home/seyyed/Downloads/cntk/DataSets/MNIST/Test-28x28_cntk_text.txt"
        input = {
            features = { dim = 784 ; format = "dense" }
            labels =   { dim = 10  ; format = "dense" }
        }
    }
}

configparameters: 01_OneHidden.cntk:traceLevel=1
configparameters: 01_OneHidden.cntk:trainNetwork={
    action = "train"
    BrainScriptNetworkBuilder = {
imageShape = 28:28:1                        
labelDim = 10                               
        featScale = 1/256
        model(x) = {
            s1 = x * featScale
            h1 = DenseLayer {200, activation=ReLU} (s1) 
            z = LinearLayer {labelDim} (h1)
        }
        features = Input {imageShape}
        labels = Input {labelDim}
        out = model (features)
        ce   = CrossEntropyWithSoftmax (labels, out.z)
        errs = ClassificationError (labels, out.z)
        featureNodes    = (features)
        labelNodes      = (labels)
        criterionNodes  = (ce)
        evaluationNodes = (errs)
        outputNodes     = (out.z)
    }
    SGD = {
        epochSize = 60000
        minibatchSize = 64
        maxEpochs = 10
        learningRatesPerSample = 0.01*5:0.005
        momentumAsTimeConstant = 0
        numMBsToShowResult = 500
    }
    reader = {
        readerType = "CNTKTextFormatReader"
        file = "/home/seyyed/Downloads/cntk/DataSets/MNIST/Train-28x28_cntk_text.txt"
        input = {
            features = { dim = 784 ; format = "dense" }
            labels =   { dim = 10  ; format = "dense" }
        }
    }   
}

Commands: trainNetwork testNetwork
precision = "float"

##############################################################################
#                                                                            #
# trainNetwork command (train action)                                        #
#                                                                            #
##############################################################################

Creating virgin network.
Node '<placeholder>' (LearnableParameter operation): Initializating Parameter[10 x 0] as glorotUniform later when dimensions are fully known.
Node '<placeholder>' (LearnableParameter operation): Initializating Parameter[200 x 0] as glorotUniform later when dimensions are fully known.

Post-processing network...

3 roots:
	ce = CrossEntropyWithSoftmax()
	errs = ClassificationError()
	out.z = Plus()

Validating network. 15 nodes to process in pass 1.

Validating --> labels = InputValue() :  -> [10 x *]
Validating --> out.z.W = LearnableParameter() :  -> [10 x 0]
Validating --> out.h1.arrayOfFunctions[0].W = LearnableParameter() :  -> [200 x 0]
Validating --> features = InputValue() :  -> [28 x 28 x 1 x *]
Validating --> _out.s1 = LearnableParameter() :  -> [1]
Validating --> out.s1 = ElementTimes (features, _out.s1) : [28 x 28 x 1 x *], [1] -> [28 x 28 x 1 x *]
Node 'out.h1.arrayOfFunctions[0].W' (LearnableParameter operation) operation: Tensor shape was inferred as [200 x 28 x 28 x 1].
Node 'out.h1.arrayOfFunctions[0].W' (LearnableParameter operation): Initializing Parameter[200 x 28 x 28 x 1] <- glorotUniform(seed=2, init dims=[200 x 784], range=0.078087(0.078087*1.000000), onCPU=true.
)Validating --> out.h1._.PlusArgs[0] = Times (out.h1.arrayOfFunctions[0].W, out.s1) : [200 x 28 x 28 x 1], [28 x 28 x 1 x *] -> [200 x *]
Validating --> out.h1.arrayOfFunctions[0].b = LearnableParameter() :  -> [200]
Validating --> out.h1._ = Plus (out.h1._.PlusArgs[0], out.h1.arrayOfFunctions[0].b) : [200 x *], [200] -> [200 x *]
Validating --> out.h1 = RectifiedLinear (out.h1._) : [200 x *] -> [200 x *]
Node 'out.z.W' (LearnableParameter operation) operation: Tensor shape was inferred as [10 x 200].
Node 'out.z.W' (LearnableParameter operation): Initializing Parameter[10 x 200] <- glorotUniform(seed=1, init dims=[10 x 200], range=0.169031(0.169031*1.000000), onCPU=true.
)Validating --> out.z.PlusArgs[0] = Times (out.z.W, out.h1) : [10 x 200], [200 x *] -> [10 x *]
Validating --> out.z.b = LearnableParameter() :  -> [10]
Validating --> out.z = Plus (out.z.PlusArgs[0], out.z.b) : [10 x *], [10] -> [10 x *]
Validating --> ce = CrossEntropyWithSoftmax (labels, out.z) : [10 x *], [10 x *] -> [1]
Validating --> errs = ClassificationError (labels, out.z) : [10 x *], [10 x *] -> [1]

Validating network. 8 nodes to process in pass 2.

Validating network, final pass.

Post-processing network complete.

attempt: Input file '/home/seyyed/Downloads/cntk/DataSets/MNIST/Train-28x28_cntk_text.txt' is not open., retrying 2-th time out of 5...
attempt: Input file '/home/seyyed/Downloads/cntk/DataSets/MNIST/Train-28x28_cntk_text.txt' is not open., retrying 3-th time out of 5...
attempt: Input file '/home/seyyed/Downloads/cntk/DataSets/MNIST/Train-28x28_cntk_text.txt' is not open., retrying 4-th time out of 5...
attempt: Input file '/home/seyyed/Downloads/cntk/DataSets/MNIST/Train-28x28_cntk_text.txt' is not open., retrying 5-th time out of 5...

[CALL STACK]
[0xcd2e49]                                                            
[0x7ff1c18dec3a]    CNTK::CNTKTextFormatReader::  CNTKTextFormatReader  (Microsoft::MSR::CNTK::ConfigParameters const&) + 0x1b3a
[0x7ff1c18c5994]                                                       + 0x16994
[0x7ff20114a025]    CNTK::ReaderShim<float>::  Init  (Microsoft::MSR::CNTK::ConfigParameters const&) + 0x415
[0x7ff201101ae4]    Microsoft::MSR::CNTK::DataReader::  DataReader  <Microsoft::MSR::CNTK::ConfigParameters>(Microsoft::MSR::CNTK::ConfigParameters const&) + 0xd24
[0xb2fb93]          std::__shared_ptr<Microsoft::MSR::CNTK::DataReader,(__gnu_cxx::_Lock_policy)2>::  __shared_ptr  <std::allocator<Microsoft::MSR::CNTK::DataReader>,Microsoft::MSR::CNTK::ConfigParameters&>(std::_Sp_make_shared_tag,  std::allocator<Microsoft::MSR::CNTK::DataReader> const&,  Microsoft::MSR::CNTK::ConfigParameters&) + 0x53
[0xb2fe69]          std::shared_ptr<Microsoft::MSR::CNTK::DataReader>  CreateObject  <Microsoft::MSR::CNTK::DataReader>(Microsoft::MSR::CNTK::ConfigParameters const&,  wchar_t const*) + 0x289
[0xb45427]          void  DoTrain  <Microsoft::MSR::CNTK::ConfigParameters,float>(Microsoft::MSR::CNTK::ConfigParameters const&) + 0x327
[0xa7e1dc]          void  DoCommands  <float>(Microsoft::MSR::CNTK::ConfigParameters const&,  std::shared_ptr<Microsoft::MSR::CNTK::MPIWrapper> const&) + 0x125c
[0x9faf71]          wmainOldCNTKConfig  (int,  wchar_t**)              + 0xc31
[0x9fdc7b]          wmain1  (int,  wchar_t**)                          + 0x7b
[0x958564]          main                                               + 0xd4
[0x7ff1fe96b830]    __libc_start_main                                  + 0xf0
[0x9a1aa9]          _start                                             + 0x29
EXCEPTION occurred: CNTKTextFormatReader: While reading '/home/seyyed/Downloads/cntk/DataSets/MNIST/Train-28x28_cntk_text.txt': Input file '/home/seyyed/Downloads/cntk/DataSets/MNIST/Train-28x28_cntk_text.txt' is not open.
seyyed@seyyed-Lenovo-Y720-15IKB:~/Downloads/cntk/cntk/bin$ 
`
### And the content of the dataset path:

`seyyed@seyyed-Lenovo-Y720-15IKB:~/Downloads/cntk/Examples/Image/DataSets/MNIST$ ls
install_mnist.py  mnist_utils.py  __pycache__  README.md  Test-28x28_cntk_text.txt  Train-28x28_cntk_text.txt

`

**Comment by sjsy** [comment]
The path of the datasets in configFile is wrong.
I run the file by overwriting it as follows:

`seyyed@seyyed-Lenovo-Y720-15IKB:~/Downloads/cntk/cntk/bin$ ./cntk configFile=~/Downloads/cntk/Examples/Image/GettingStarted/01_OneHidden.cntk RootDir=/home/seyyed/Downloads/cntk dataDir=/home/seyyed/Downloads/cntk/Examples/Image/DataSets/MNIST 
`
